src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessService.java

```diff
@@ -60,21 +66,48 @@
 
     public void handleEvents(List<GoCardlessEvent> events) {
         events.forEach(goCardlessService::storeEvent);
+        updateStatesForEvents(events);
+        var mapOfBooleanToListOfEvents = events.stream().collect(partitioningBy(event -> shouldBeHandled(event)));
+        handleValidEvents(mapOfBooleanToListOfEvents.get(true));
+        handleInvalidEvents(mapOfBooleanToListOfEvents.get(false));
+    }
 
+    private void updateStatesForEvents(List<GoCardlessEvent> events) {
         Map<GoCardlessResourceType, List<GoCardlessEvent>> eventsGroupedByResourceType = events.stream().collect(groupingBy(GoCardlessEvent::getResourceType));
-        updateStateForMandateEvents(eventsGroupedByResourceType.getOrDefault(GoCardlessResourceType.MANDATES, Collections.emptyList()));
-        updateStateForPaymentEvents(eventsGroupedByResourceType.getOrDefault(GoCardlessResourceType.PAYMENTS, Collections.emptyList()));
+        updateStateForMandateEvents(eventsGroupedByResourceType.getOrDefault(MANDATES, Collections.emptyList()));
+        updateStateForPaymentEvents(eventsGroupedByResourceType.getOrDefault(PAYMENTS, Collections.emptyList()));
+    }
+
+    private void handleInvalidEvents(List<GoCardlessEvent> events) {
+//        LOGGER.info("unhandled resource type for event with id {} ", event.getGoCardlessEventId());
+    }
 
+    private void handleValidEvents(List<GoCardlessEvent> events) {
         events.forEach(event -> {
-            GoCardlessActionHandler handler = getHandlerFor(event.getResourceType());
             LOGGER.info("About to handle event of type: {}, action: {}, resource id: {}",
                     event.getResourceType(),
                     event.getAction(),
                     event.getResourceId());
-            handler.handle(event);
+            
+            if (event.getResourceType() == MANDATES) {
```

Comment:
We could do something like the `groupingBy` we do for updating states to avoid having to do this check for every event

Reply:
fixed